go.mod

```diff
@@ -0,0 +1,5 @@
+module github.com/google/flatbuffers
```

Comment:
does this file have to sit in the root?

Reply:
Yes, 
Refer to https://github.com/google/flatbuffers/pull/5741#discussion_r371401105

Reply:
You can move the `go.mod` into `/go` directory and change the mod file to:

```
module github.com/google/flatbuffers/go
```

Reply:
> You can move the `go.mod` into `/go` directory and change the mod file to: 
`module github.com/google/flatbuffers/go`

yes but for example,

```golang
// see file ./tests/go_test.go
import (
   mygame "github.com/google/flatbuffers/tests/MyGame"          // refers to generated code
	 example "github.com/google/flatbuffers/tests/MyGame/Example" // refers to generated code
)
```
and 
```golang
// see file ./samples/sample_binary.go
import sample "github.com/google/flatbuffers/samples/MyGame/Sample"
```

wont work anymore... 


Yes, I can add yet even more go.mod files in each of ./grpc, ./sample, ./test. 
But I dont think that's idiomatic. 

My general thinking is [package.json](https://github.com/google/flatbuffers/blob/master/package.json) for Javascript is in the root... [composer.json](https://github.com/google/flatbuffers/blob/master/composer.json)  for PHP is in the root etc etc. So I think let's just put go.mod in the root as it's serving a similar purpose (and, as far as I know, it's not unidiomatic to do so)

Reply:
For `go`, grpc code is stored inside `/go` folder.

For the tests, you can create the go.mod before running them and delete it after.

Reply:
yes grpc code is in ./go
but *grpc tests* are in ./grpc
and go sample code is in ./sample

> For the tests, you can create the go.mod before running them and delete it after.

but why do all that?

The point is when a Go contributor used their IDE to even just casually open go code in places other than ./go, they shouldn't get the ugly red underlinings under the import statements. And because Go doesn't yet understand the import statements, the contributors will surely get those red underlinings if go.mod is only ever created when ./tests/GoTest.sh is run, and then deleted after.

Reply:
yes grpc code is in ./go
but *grpc tests* are in ./grpc
and go sample code is in ./sample

> For the tests, you can create the go.mod before running them and delete it after.

but why do all that?

The point is when a Go contributor used their IDE to even just casually open go code in places other than ./go, they shouldn't get the ugly red underlinings under the import statements. And because Go doesn't yet understand the import statements, the contributors will surely get those red underlinings if go.mod is only ever created when ./tests/GoTest.sh is run, and then deleted after.